app/controllers/people_controller.rb

```diff
@@ -19,7 +19,7 @@ class PersonDeleted < StandardError; end
   helper_method :show_closed?
 
   def show
-    @person = Person.find(params[:person_id] || params[:id])
+    @person = Person.find_by_username_and_community_id!(params[:username], @current_community.id)
     raise PersonDeleted if @person.deleted?
     PersonViewUtils.ensure_person_belongs_to_community!(@person, @current_community)
```

Comment:
This can be now removed because `Person.find_by_username_and_community_id!` already ensures that the person belongs to community.